snuba/datasets/storages/factory.py

```diff
@@ -55,6 +59,27 @@
 from snuba.datasets.storages.transactions import storage as transactions_storage
 from snuba.datasets.storages.transactions_ro import storage as transactions_ro_storage
 from snuba.datasets.storages.transactions_v2 import storage as transactions_v2_storage
+from snuba.state import get_config
+
+gen_metrics_dists_bucket_storage = gen_metrics_dists_bucket_storage_original
+gen_metrics_dists_aggregate_storage = gen_metrics_dists_aggregate_storage_original
+gen_metrics_sets_bucket_storage = gen_metrics_sets_bucket_storage_original
+gen_metrics_sets_aggregate_storage = gen_metrics_sets_aggregate_storage_original
+
+if get_config("use_generic_metrics_storages_from_configs", 0):
+    gen_metrics_dists_bucket_storage = build_writable_storage(
+        StorageKey.GENERIC_METRICS_DISTRIBUTIONS_RAW
+    )
+    gen_metrics_dists_aggregate_storage = build_readonly_storage(
```

Comment:
given that we have the `kind` value in the config, why don't we use that to determine what kind of storage we're building, and then everything can be loaded from one generic entrypoint loader?

Reply:
I had that before and there were messy typing issues when returning `ReadableTableStorage | WritableTableStorage` so I split it into 2 functions. I guess since these will be automatically loaded I do have to go back to doing it that way, I'll change it.